add noBias for nn.Linear and nn.SpatialConvolution

yangky11 opened the pull request:
Hi,

This is to allow `nn.Linear` and `nn.SpatialConvolution` to have no bias, just like [cudnn.torch](https://github.com/soumith/cudnn.torch).

We can revert https://github.com/soumith/cudnn.torch/pull/164 if this is merged.

@fmassa

yangky11 commented:
Somehow a test on `VolumetricMaxPooling` failed, but it seems has nothing to do with the new changes.

fmassa commented:
Current implementation allocates new memory at each forward/backward if the no-bias option is set.
I have started the no-bias option some time ago, but I was wanting to factorize the convolution into smaller blocks as well and I didn't finish it. It's here if you want to have a look https://github.com/fmassa/nn/commit/6a6f5f155edd1e3aa943ddb0f0f91ee79d9d5257

yangky11 commented:
Thanks, I just integrated your code into this PR so that it does not allocate new memories, added some tests, and removed the unnecessary argument `bias` in `updateGradInput`

fmassa commented:
Cool!
Could you do the same for cunn (as is cunn won't work) ? I also have the changes somewhere in my computer, just need to check where they are.

yangky11 commented:
I'm working on it

soumith commented:
Thanks Kaiyu!
